app.py

```diff
@@ -0,0 +1,110 @@
+import os
+import json
+import openai
+from target import question_answer_fast
+from answering import answer_input_questions
+from advance_search import question_answer_prompts, question_answer_prompting
+from container.src.validation import load_credentials
+from flask import Flask, redirect, render_template, request, url_for
+
+
+count = 0
+app = Flask(__name__)
+
+
+@app.route("/inputQuestions/", methods=("GET", "POST"))
+def inputQuestionsAPI():
+    if type(request.json) == str:
+        try:
+            data = json.loads(request.json)
+        except:
+            print("Invalid request parameter provided. Please use json object")
+    else:
+        data = request.json
+    
+    print("Data recieved with fields:", list(data.keys()))
+    # ToDo: Refactor this so it finds the function from a map instead of with if statements
+    if request.method == "POST":
+        # Not using .get method to keep dataType consistent
+        id = ""
+        if "id" in data:
+            id = data["id"]
+        
+        # Specifies a functionality for the request
+        operation = ""
+        if "operation" in data:
+            operation = data["operation"]
+
+        # Parses the pages innerHTML and finds the questions and answer input elements
+        # Uses LLM prompting
+        if operation == "Question-Answer-LLM":
+            # This functionality is the following two operations combined
+            htmlContent = data["requestData"]
+            prompts = question_answer_prompts(id, htmlContent)
+            responses = question_answer_prompting(id, prompts)
+            return json.dumps(qas)
+        
+        if operation == "Question-Answer-Prompts":
+            # "Question-Answer-LLM" but only returns the prompts
+            htmlContent = data["requestData"]
+            prompts = question_answer_prompts(id, htmlContent)
+            return json.dumps(prompts)
+        
+        if operation == "Question-Answer-Prompting":
+            # "Question-Answer-LLM" but only returns the prompts
+            prompts = data["requestData"]
+            responses = question_answer_prompting(id, prompts)
+            return json.dumps(responses)
+        
+        # Parses the pages innerHTML and finds the questions and answer input elements
+        # Uses same parent element to find question answer pairs, must faster
+        if operation == "Question-Answer-Fast":
```

Comment:
Minor nit: For cases where you are matching an operation, you  should use `elif`.

Alternatively, for Python 3.10+, try case switches https://www.freecodecamp.org/news/python-switch-statement-switch-case-example/

Reply:
I was planning to replace the if statements with a function map. Its implemented.